[IMP] http: improve dumpstacks and remaining requests handling during tests

amigrave opened the pull request:
The patch modifies Root#dispatch() in order to keep the httprequest url
in the current thread's attribute and use if for dumpstacks and the
phantomjs tests remaining requests handling.

Also, stop looping forever when trying to join unclosed http requests
in phantomjs test suite after a phantomjs failure

odony commented on odoo/http.py:
```diff
@@ -1440,6 +1440,7 @@ def dispatch(self, environ, start_response):
             httprequest = werkzeug.wrappers.Request(environ)
             httprequest.app = self
             httprequest.parameter_storage_class = werkzeug.datastructures.ImmutableOrderedMultiDict
+            threading.current_thread().httprequest_url = httprequest.url
```
Doesn't this require a corresponding cleanup in `application_unproxied()`?
https://github.com/odoo/odoo/blob/10.0/odoo/service/wsgi_server.py#L157-L165

amigrave commented:
@odony I cleaned the property as you proposed. I also renamed httpresquest_url to url and I increased the delay and decreased the counter in the join retry loop as requested by @antonylesuisse

odony commented:
@amigrave :+1: for all these changes!
